### PR TITLE
Testing enhancements

### PR DESCRIPTION
**Pull request by doomspork**
The goal of this PR is multifaceted: 

1. Make testing easier by shifting the dependency on content to a smaller, faster to compile, and known set of content. This collection can grow to enable better testing but can remain much smaller in size than the total collection.
2. Add test coverage for the modules implementing NimblePublisher and handling our content.

### CODE REVIEW

File: lib/school_house/posts.ex [hunk 1]
```diff
@@ -4,18 +4,17 @@ defmodule SchoolHouse.Posts do
...
-    from: "content/posts/**/*.md",
+    from: Application.compile_env!(:school_house, :blog_dir),
     as: :posts,
     highlighters: [:makeup_elixir, :makeup_erlang]
 
   @paged_posts @posts |> Enum.sort(&(Date.compare(&1.date, &2.date) == :gt)) |> Enum.chunk_every(20)
   @posts_by_slug Enum.into(@posts, %{}, fn %{slug: slug} = post ->
                    {slug, post}
                  end)
+  def get(slug), do: Map.get(@posts_by_slug, slug)
```
Comment (doomspork) [inline lib/school_house/posts.ex:15]: Reordered the functions based on :abc:

**Comment by doomspork** [comment]
@notactuallypagemcconnell here's what I came up with. I leaned on `Application.compile_env!/3` here to fetch the content path during compilation. This lets me point to a fixture directory :tada:

Once GitHub Actions are unborked I'll re-run this. Locally all 3 checks pass.

**Comment by doomspork** [comment]
Thoughts on moving some of these into doctests?

**Review by notactuallytreyanastasio — APPROVED** [review]
Looks great, by the book dependency injection ftw 🥳

**Comment by doomspork** [comment]
Ah, your other Bobby user must still be the contributor to ES :laughing: Gotta fix that at some point.

**Comment by doomspork** [comment]
@gemantzu / @jaimeiniesta want to take a peek at this when you have a chance? I'll need someone else to approve :+1:

**Review by gemantzu — APPROVED** [review]
:fire:

**Pull request merged**